data/expenses.js

```diff
@@ -0,0 +1,152 @@
+const { ObjectId, ObjectID } = require('mongodb');
+const { expenses: getExpensesCollection } = require('../config/mongoCollections');
+
+const { QueryError, ValidationError } = require('../utils/errors');
+const { idQuery, parseMongoData, stringifyObjectId } = require('../utils/mongodb');
+const {
+  assertObjectIdString,
+  assertIsValuedString,
+  assertRequiredObject,
+} = require('../utils/assertion');
+
+const getByObjectId = async (objectId) => {
+  const collection = await getExpensesCollection();
+  const expense = await collection.findOne(idQuery(objectId));
+  return parseMongoData(expense);
+};
+
+const getExpense = async (expenseId) => {
+  assertObjectIdString(expenseId, 'Expense id');
+
+  let expense = await getByObjectId(expenseId);
+  if (expense == null) {
+    throw new QueryError(`Could not get expense for (${expenseId})`);
+  }
+  return expense;
+};
+
+const getAllExpenses = async (tripId) => {
+  assertObjectIdString(tripId, 'Trip id');
+
+  //const trip = await getTrip(tripId);
+
+  const collection = await getExpensesCollection();
+  const allExpenses = await collection.find({ tripId: new ObjectId(tripId) }).toArray();
+  if (allExpenses.length == 0) {
+    throw new QueryError(`Expenses not found`);
+  }
+
+  return allExpenses;
```

Comment:
If you're using `parseMongoData`, it goes for lists, too. Here, the return values will not have `ObjectId`s as strings as your other functions.

Reply:
I have changed and now we are getting a string.

Reply:
Actually, you can just put anything(array, object, string) into `parseMongoData`, it basically finds and stringifies `ObjectId` values for you.